modin/pandas/indexing.py

```diff
@@ -293,28 +311,26 @@ def _compute_enlarge_labels(self, locator, base_index):
             )
         return nan_labels
 
-    def _expand_dim(self, row_lookup, col_lookup, ndim):
-        """Expand the dimension if necessary.
-        This method is for cases like duplicate labels.
-        """
-        many_rows = len(row_lookup) > 1
-        many_cols = len(col_lookup) > 1
-
-        if ndim == 0 and (many_rows or many_cols):
-            ndim = 1
-        if ndim == 1 and (many_rows and many_cols):
-            ndim = 2
-        return ndim
-
     def _compute_lookup(self, row_loc, col_loc) -> Tuple[pandas.Index, pandas.Index]:
-        if isinstance(row_loc, list) and len(row_loc) == 1:
+        if is_list_like(row_loc) and len(row_loc) == 1:
             if (
                 isinstance(self.qc.index.values[0], np.datetime64)
                 and type(row_loc[0]) != np.datetime64
             ):
                 row_loc = [pandas.to_datetime(row_loc[0])]
-        row_lookup = self.qc.index.to_series().loc[row_loc].index
-        col_lookup = self.qc.columns.to_series().loc[col_loc].index
+
+        if isinstance(row_loc, slice):
+            row_lookup = self.qc.index.to_series().loc[row_loc].values
+        elif isinstance(self.qc.index, pandas.MultiIndex):
```

Comment:
Should we put the `MultiIndex` check before https://github.com/modin-project/modin/blob/7a1391adaebd8b5f67a61c6a2a6ae667204567f0/modin/pandas/indexing.py#L335

Reply:
No, because `get_locs` does not work with a `slice`, so we have to check and evaluate the slice first.

Reply:
ah ok thanks!